Sources/SwiftParser/IncrementalParseTransition.swift

```diff
@@ -48,25 +48,39 @@ public typealias ReusedNodeCallback = (_ node: Syntax) -> ()
 /// Keeps track of a previously parsed syntax tree and the source edits that
 /// occurred since it was created.
 public final class IncrementalParseTransition {
-  fileprivate let previousTree: SourceFileSyntax
+  fileprivate let previousTreeWithLookahed: SyntaxTreeWithLookahead
   fileprivate let edits: ConcurrentEdits
-  fileprivate let lookaheadRanges: LookaheadRanges
   fileprivate let reusedNodeCallback: ReusedNodeCallback?
 
   /// - Parameters:
   ///   - previousTree: The previous tree to do lookups on.
   ///   - edits: The edits that have occurred since the last parse that resulted
   ///            in the new source that is about to be parsed.
   ///   - reusedNodeCallback: Optional closure to accept information about the re-used node. For each node that gets re-used `reusedNodeCallback` is called.
+  @available(*, deprecated, message: "Use initializer taking `SyntaxTreeWithLookahead` instead")
   public init(
     previousTree: SourceFileSyntax,
     edits: ConcurrentEdits,
     lookaheadRanges: LookaheadRanges,
     reusedNodeCallback: ReusedNodeCallback? = nil
   ) {
-    self.previousTree = previousTree
+    self.previousTreeWithLookahed = SyntaxTreeWithLookahead(tree: previousTree, lookaheadRanges: lookaheadRanges)
+    self.edits = edits
+    self.reusedNodeCallback = reusedNodeCallback
+  }
+
+  /// - Parameters:
+  ///   - previousTreeWithLookahed: The previous tree to do lookups on.
+  ///   - edits: The edits that have occurred since the last parse that resulted
+  ///            in the new source that is about to be parsed.
+  ///   - reusedNodeCallback: Optional closure to accept information about the re-used node. For each node that gets re-used `reusedNodeCallback` is called.
+  public init(
+    previousTreeWithLookahed: SyntaxTreeWithLookahead,
```

Comment:
If we rename `SyntaxTreeWithLookahead` -> `IncrementalParseResult`, I think the parameter/variable should be called `previousIncrementalParseResult`.

Reply:
Totally agree 👍